functional/pipeline.py

```diff
@@ -265,9 +266,10 @@ def head_option(self):
 
         :return: first element of sequence or None if sequence is empty
         """
-        if not self.sequence:
+        try:
```

Comment:
I think I agree with this change, but for a different reason. `self.sequence` calls and caches the result of `to_list()` when in reality it should not be necessary to evaluate the entire list just to do `head_option`. `head` is written such that it does not need to do this so calling it gains the benefit of that.

```
In [1]: from functional import seq

In [2]: a = seq.range(10)

In [4]: def f(x):
   ...:     print('eval', x)
   ...:     return x
   ...:

In [5]: a.map(f).head()
eval 0
Out[5]: 0

In [6]: a = seq.range(10)

In [7]: a.map(f).head_option()
eval 0
eval 1
eval 2
eval 3
eval 4
eval 5
eval 6
eval 7
eval 8
eval 9
Out[7]: 0
```